Function SALib.util_funcs.avail_approaches fails when all tests are run, passes in isolation

# Fails when the suite is run
The test `test_cli.py:test_cli_avail_methods` fails when calling `SALib.util_funcs.avail_approaches` with the error:
```python
________________________________________________________ test_cli_avail_methods ________________________________________________________

    def test_cli_avail_methods():
        method_types = ['sample', 'analyze']
    
        for method in method_types:
            module = importlib.import_module('.'.join(['SALib', method]))
            actions = avail_approaches(module)
            for act in actions:
>               approach = importlib.import_module('.'.join(
                    ['SALib', method, act]))

tests/test_cli.py:28: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../miniconda3/envs/salib/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1014: in _gcd_import
    ???
<frozen importlib._bootstrap>:991: in _find_and_load
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

name = 'SALib.sample.morris.test_morris', import_ = <function _gcd_import at 0x7fdf645884c0>

>   ???
E   ModuleNotFoundError: No module named 'SALib.sample.morris.test_morris'

<frozen importlib._bootstrap>:973: ModuleNotFoundError
```
but only when all tests are run together. 
# Passes when run in isolation
When `pytest tests/test_cli.py` is run, the tests pass fine.
```python
$ pytest tests/test_cli.py 
========================================================= test session starts ==========================================================
platform darwin -- Python 3.8.2, pytest-5.4.1, py-1.8.1, pluggy-0.13.1 -- /Users/wusher/miniconda3/envs/salib/bin/python
cachedir: .pytest_cache
rootdir: /Users/wusher/repository/SALib, inifile: setup.cfg
plugins: cov-2.8.1
collected 2 items                                                                                                                      

tests/test_cli.py::test_cli_usage PASSED                                                                                         [ 50%]
tests/test_cli.py::test_cli_avail_methods PASSED                                                                                 [100%]Coverage.py warning: No data was collected. (no-data-collected)
WARNING: Failed to generate report: No data to report.

/Users/wusher/miniconda3/envs/salib/lib/python3.8/site-packages/pytest_cov/plugin.py:254: PytestWarning: Failed to generate report: No data to report.

  self.cov_controller.finish()


=========================================================== warnings summary ===========================================================
tests/test_cli.py::test_cli_avail_methods
  <frozen importlib._bootstrap>:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject

-- Docs: https://docs.pytest.org/en/latest/warnings.html

---------- coverage: platform darwin, python 3.8.2-final-0 -----------

===================================================== 2 passed, 1 warning in 6.54s =====================================================
```

--- a/src/SALib/util/util_funcs.py
+++ b/src/SALib/util/util_funcs.py
@@ -21,7 +21,9 @@ def avail_approaches(pkg):
     methods = [modname for importer, modname, ispkg in
                pkgutil.walk_packages(path=pkg.__path__)
                if modname not in
-               ['common_args', 'directions', 'sobol_sequence']]
+               ['common_args', 'directions', 'sobol_sequence']
+               and 'test' not in modname]
+
     return methods
 
 

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -22,11 +22,10 @@ def test_cli_avail_methods():
     method_types = ['sample', 'analyze']
 
     for method in method_types:
-        module = importlib.import_module('.'.join(['SALib', method]))
+        module = importlib.import_module(f'SALib.{method}')
         actions = avail_approaches(module)
         for act in actions:
-            approach = importlib.import_module('.'.join(
-                ['SALib', method, act]))
+            approach = importlib.import_module(f'SALib.{method}.{act}')
 
             # Just try to access the functions - raises error on failure
             approach.cli_parse

